discovery/src/main/scala/akka/discovery/SimpleServiceDiscovery.scala

```diff
@@ -56,28 +56,32 @@ object SimpleServiceDiscovery {
       }
   }
 
-  /**
-   * Represents a service discovery lookup.
-   *
-   * Each mechanism should document what a Simple vs Full lookup means.
-   * Implementations are free to ignore port/protocol from a Full lookup
-   * but should not throw if they are set unexpectedly as this prevents the
-   * same query being sent to multiple mechanisms via the aggregate service
-   * discovery
-   */
-  @ApiMayChange
-  sealed abstract class Lookup {
-    def name: String
-  }
+}
 
-  @ApiMayChange
-  final case class Simple(name: String) extends Lookup
+/**
+ * Additional information for a service lookup. It is up to each mechanism to decide
+ * what to do with this.
+ */
+@ApiMayChange
+class LookupMetadata(portName: Option[String] = None, protocol: Option[String] = None) {
```

Comment:
uhm... Don't be angry but isn't this a bit weird...?
A normal Lookup(name, portname, protocol) would have been fine I thought?

Reply:
This makes it a bit harder to pass around something to be idenfitied -- "the thing" is IDd after all using name+portName+protocol for SVC.

If you really love this idea we can keep it but what I thought we had agreed on was the `Lookup(name, opt port name, opt protocol)`